### PR TITLE
src: rework includes as per cppclean

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
Is there already one @jasonish ?

Describe changes:
- Rework includes as per cppclean 

Still to do : add `cppclean` as part of CI

I used this patched version https://github.com/myint/cppclean/pull/165

There are still exceptions :
- About Suricata : When "rust.h" or "suricata-common.h" gets included, and when includes is rust.h or rust-context.h
- About cppclean 
  - when using macros such as `TAILQ` with including queue.h (or red and black tree)
  - when using the array size, such as ``uint8_t plugin_v[PLUGIN_VAR_SIZE];  in decode.h needing suricata-plugin.h

The result is less includes from header file to another, nothing is done for C files including a useless header

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7955

**Comment by jasonish** [comment]
> Is there already one @jasonish ?

No, any include cleanups I've done in the past have been related to other fixups, so no specific include cleanup ticket yet.

**Comment by jasonish** [comment]
Would be great if we could get this to work, as in remove all unecessary includes from headers, then source files. Would make it much easier to proceed with some manual cleanup, and logical organization with respect to libsuricata and what we consider private and public.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7962

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7965

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7967

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7970

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7973

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 7974

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ips_afp_drop_chk.
  

Pipeline 7975

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ips_afp_drop_chk.
  

Pipeline 7979

File: src/app-layer-nbss.h [hunk 1]
```diff
@@ -24,11 +24,6 @@
 #ifndef __APP_LAYER_NBSS_H__
```
Comment (victorjulien) [inline src/app-layer-nbss.h:24]: looks like this file can be removed instead? Should have been removed as part of the smb to rust work years ago I think

**Comment by victorjulien** [comment]
Looks good to me. Will run in my own ci setup as well to get a few more build types

**Comment by victorjulien** [comment]
```
  CC       util-mpm-hs.o
util-mpm-hs.c: In function 'SCHSTest29':
util-mpm-hs.c:2118:5: error: implicit declaration of function 'SigGroupBuild' [-Werror=implicit-function-declaration]
 2118 |     SigGroupBuild(de_ctx);
      |     ^~~~~~~~~~~~~
util-mpm-hs.c:2134:9: error: implicit declaration of function 'SigGroupCleanup' [-Werror=implicit-function-declaration]
 2134 |         SigGroupCleanup(de_ctx);
      |         ^~~~~~~~~~~~~~~
util-mpm-hs.c:2135:9: error: implicit declaration of function 'SigCleanSignatures'; did you mean 'SigLoadSignatures'? [-Werror=implicit-function-declaration]
 2135 |         SigCleanSignatures(de_ctx);
      |         ^~~~~~~~~~~~~~~~~~
      |         SigLoadSignatures
```

File: src/app-layer-nbss.h (hunk 1 above)
Comment (catenacyber) [reply src/app-layer-nbss.h:24]: Oh and so are decode-null.h and util-vector.h :-)

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7570 fixing util-mpm-hs.c to include detect-engine-build.h (and also in detect-dce-iface.c detect-dce-opnum.c tests/detect.c)

**Pull request closed without merging**